src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/DocumentConversionService.java

```diff
@@ -85,6 +92,41 @@ public byte[] flattenPdfDocument(byte[] document) {
         return document;
     }
 
+    private void flattenAnnotations(PDDocument doc) throws IOException {
+        for (PDPage page : doc.getPages()) {
+            List<PDAnnotation> annotations = page.getAnnotations();
+            for (PDAnnotation annotation : annotations) {
+                PDAppearanceStream appearanceStream = annotation.getNormalAppearanceStream();
+                if (appearanceStream != null) {
+                    transformAnnotationsToContentStream(doc, page, annotation, appearanceStream);
+                }
+            }
+            annotations.clear();
+        }
+    }
+
+    private void transformAnnotationsToContentStream(PDDocument doc, PDPage page, PDAnnotation annotation,
+                                                     PDAppearanceStream appearanceStream) throws IOException {
+        float appearanceWidth = (float) appearanceStream.getBBox().getWidth();
+        float appearanceHeight = (float) appearanceStream.getBBox().getHeight();
+
+        if (appearanceWidth > 0 && appearanceHeight > 0) {
+            try (PDPageContentStream contentStream =
+                     new PDPageContentStream(doc, page, PDPageContentStream.AppendMode.APPEND,
+                         true, true)) {
+                contentStream.saveGraphicsState();
+                float x = (float) annotation.getRectangle().getLowerLeftX();
+                float y = (float) annotation.getRectangle().getLowerLeftY();
+                float width = (float) annotation.getRectangle().getWidth();
+                float height = (float) annotation.getRectangle().getHeight();
+                Matrix transformation = new Matrix(width / appearanceWidth, 0, 0, height / appearanceHeight, x, y);
+                contentStream.transform(transformation);
+                contentStream.drawForm(appearanceStream);
+                contentStream.restoreGraphicsState();
+            }
+        }
+    }
```

Comment:
Got AI to explain this to me, and it seems correct.  However, I would include some plain english JDOC, just to give a flavour on what is happening here, for future readers.